fastlane/lanes/localization.rb

```diff
@@ -122,49 +127,78 @@
-
-    # Create a temporary directory to clone Gutenberg into.
-    # We'll run the rest of the automation from within the block, but notice that only the Gutenbreg cloning happens within the temporary directory.
-    gutenberg_clone_name = 'Gutenberg-Strings-Clone'
+    # We jump in the tempdir immediately, even though we might not need it.
+    # It's just simpler to rely on the Ruby block API for it than to manage it by hand.
     Dir.mktmpdir do |tempdir|
-      Dir.chdir(tempdir) do
-        repo_url = "https://github.com/#{github_org}/#{repo_name}"
-        UI.message("Cloning Gutenberg from #{repo_url} into #{gutenberg_clone_name}. This might take a few minutes…")
-        sh("git clone --depth 1 #{repo_url} #{gutenberg_clone_name}")
-        Dir.chdir(gutenberg_clone_name) do
-          if config[:ref][:tag]
-            sh("git fetch origin refs/tags/#{ref}:refs/tags/#{ref}")
-            sh("git checkout refs/tags/#{ref}")
-          else
-            sh("git fetch origin #{ref}")
-            sh("git checkout #{ref}")
+      # Fetch fresh pods to read the latest localizations from them.
+      # In CI, we expect the pods to be already available and up to date.
+      cocoapods unless is_ci
+
+      # For the same reason, fetch fresh packages
+      # FIXME: We yet don't have an explicit way to set the derived data in CI, so we'll run it in CI, too
+      derived_data_path = options[:derived_data_path] || DERIVED_DATA_PATH
+      resolve_packages(derived_data_path: derived_data_path)
+
+      custom_gutenber_path = options[:gutenberg_path]
+      if custom_gutenber_path
+        # It's simpler to ask the caller to give an absolute path than to make it absolute ourselves be
+        # Given this is a debug option, it seems like an okay tradeoff.
+        UI.user_error!("Please provide gutenberg_path as an absolute path. Current value: #{custom_gutenber_path}") unless File.absolute_path?(custom_gutenber_path)
+
+        UI.user_error!("Could not find Gutenber project at given path #{custom_gutenber_path}") unless Dir.exist?(custom_gutenber_path)
+
+        gutenberg_path = custom_gutenber_path
+        UI.message("Using Gutenberg from #{gutenberg_path} instead of cloing it...")
+      else
+        # On top of fetching the latest Pods, we also need to fetch the source for the Gutenberg code.
+        # To get it, we need to manually clone the repo, since Gutenberg is distributed via XCFramework.
+        # XCFrameworks are binary targets and cannot extract strings via genstrings from there.
+        config = gutenberg_config!
+
+        ref_node = config[:ref]
+        UI.user_error!('Could not find Gutenberg ref to clone the repository in order to access its strings.') if ref_node.nil?
+
+        ref = ref_node[:tag] || ref_node[:commit]
+        UI.user_error!('The ref to clone Gutenberg in order to access its strings has neither tag nor commit values.') if ref.nil?
```

Comment:
❓ Just to confirm: do we want to support `ref_node[:branch]` as well or not?

I'm guessing no, because I wouldn't recommend pointing to a branch to begin with—as those can disappear at any point once the corresponding PR is merged and GitHub will delete it—but still asking just to be sure we're in agreement with the needs.

Reply:
Confirming that I don't think we should support reading from a branch for the very same rationale you gave.

Semi-related internal discussion pbArwn-6DH-p2